src/calculated-property-rule.ts

```diff
@@ -108,31 +108,36 @@ export class CalculatedPropertyRule extends Rule {
 
 		// modify list properties to match the calculated value instead of overwriting the property
 		if (this.property.isList) {
-			// re-calculate the list values
-			var newList = newValue;
+			if (!this.property.isInited(obj)) {
+				Property$init(this.property, obj, newValue);
+			}
+			else {
+				// re-calculate the list values
+				var newList = newValue;
 
-			// compare the new list to the old one to see if changes were made
-			var curList = this.property.value(obj) as ObservableArray<any>;
```

Comment:
This would cause infinite recursion because setting the calculated value for a list required obtaining the value, which would cause the rule to rerun, which would bring us back here.

Reply:
@tskimmett Do you know if the same behavior exists in the legacy codebase? If not, since this code is the same, I wonder what the difference is (maybe a missing short-circuit somewhere)?

Reply:
I'm not sure if it exists in legacy. I don't think we have any way to generate calculated list properties in the product though. I only discovered this issue when writing a test.

Reply:
Ah, good point. Only thing I could think of is lookups, but I'm guessing you're not using calculations for that, but manually managing lists?

Reply:
Right, lookups can't be calculated (yet,, ;)